core/focus_manager.ts

```diff
@@ -89,7 +90,16 @@ export class FocusManager {
       }
 
       if (newNode) {
-        this.focusNode(newNode);
+        const newTree = newNode.getFocusableTree();
+        const oldTree = this.focusedNode?.getFocusableTree();
+        if (newNode === newTree.getRootFocusableNode() && newTree !== oldTree) {
```

Comment:
Note to reviewer: this is a functionality change from before. These changes mean that tabbing to the root of a tree will always behave the same as focusTree(), that is, it will restore focus active focus back to the previous node if there was any. This introduces consistency with explicit focusTree() calls, but may not always be expected by the user as it will shift focus to something new. Note that the extra tree check is to ensure users can tab out of a tree (in which case they WILL go to the root and set that as the last known focus state of the tree which should generally stabilize the user experience since they explicitly navigated away).

Reply:
Assuming approval means this behavior change is okay. :)

Reply:
Yes. The behaviour change makes sense and is well explained.